maven-core/src/main/java/org/apache/maven/project/ProjectSorter.java

```diff
@@ -126,43 +123,29 @@ public ProjectSorter( Collection<MavenProject> projects )
                          parent.getVersion(), true, false );
             }
 
-            List<Plugin> buildPlugins = project.getBuildPlugins();
-            if ( buildPlugins != null )
```

Comment:
The old code suggested that `null` might be returned from `project.getBuildPlugins()`. Is that no longer true?

Reply:
Yes I know that the old code implied that there might be `null` an option but If you take a look into `project.getBuildPlugins();`
```java
    public List<Plugin> getBuildPlugins()
    {
        if ( getModel().getBuild() == null )
        {
            return Collections.emptyList();
        }
        return Collections.unmodifiableList( getModel().getBuild().getPlugins() );
    }
```
furthermore the part: `getModel().getBuild().getPlugins()`:
```java
    public java.util.List<Plugin> getPlugins()
    {
        if ( this.plugins == null )
        {
            this.plugins = new java.util.ArrayList<Plugin>();
        }

        return this.plugins;
    } //-- java.util.List<Plugin> getPlugins()
```
So can't be returning `null`...so the check does not make sense anymore.